engine/access/rpc/backend/connection_factory.go

```diff
@@ -84,6 +111,20 @@ func (cf *ConnectionFactoryImpl) createConnection(address string, timeout time.D
 		Timeout: timeout,
 	}
 
+	var connInterceptors []grpc.UnaryClientInterceptor
+
+	cbInterceptor := cf.createCircuitBreakerInterceptor()
+	if cbInterceptor != nil {
+		connInterceptors = append(connInterceptors, cbInterceptor)
+	}
+
+	ciInterceptor := cf.createClientInvalidationInterceptor(address, clientType)
+	if ciInterceptor != nil {
+		connInterceptors = append(connInterceptors, ciInterceptor)
+	}
```

Comment:
how about moving the checks here instead of returning nil? it make the code clearer
```suggestion
	if cf.CircuitBreakerConfig.Enabled {
		connInterceptors = append(connInterceptors, cf.createCircuitBreakerInterceptor())
	} else {
		connInterceptors = append(connInterceptors, cf.createClientInvalidationInterceptor(address, clientType))
	}
```